### PR TITLE
[usage] Alert on Usage and Invoice Reconciliations

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Alert on the two core RPCs used in usage reconciliation. 

## Related Issue(s)

* Soft depends on https://github.com/gitpod-io/gitpod/pull/12917

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: operations/observability/mixins/meta/rules/usage.yaml [hunk 1]
```diff
@@ -14,13 +14,24 @@ spec:
...
+      expr: sum(increase(grpc_server_handled_total{grpc_service="usage.v1.UsageService", grpc_method="ReconcileUsage", grpc_code!="OK"})) > 1
       for: 30m
       labels:
         severity: warning
         team: webapp
       annotations:
-        runbook_url: https://github.com/gitpod-io/runbooks/blob/main/runbooks/GitpodUsageScheduledReconciliationFailures.md
-        summary: There are failed scheduled reconciliations in the usage component.
-        description: We have accumulated {{ printf "%.2f" $value }} failures. This affects how stale usage data is and/or updating invoices in Stripe.
+        runbook_url: https://github.com/gitpod-io/runbooks/blob/main/runbooks/GitpodUsageReconcileUsageFailures.md
```
Comment (laushinka) [inline operations/observability/mixins/meta/rules/usage.yaml:24]: Is this the new name for [this runbook](https://github.com/gitpod-io/runbooks/blob/main/runbooks/GitpodUsageScheduledReconciliationFailures.md)?

Comment (easyCZ) [reply operations/observability/mixins/meta/rules/usage.yaml:24]: Yeah I'll need to update the runbooks correspondingly. For now these would only be Slack warnings in webapp so we've some time to polish the runbooks before we get it to actually page

Comment (easyCZ) [reply operations/observability/mixins/meta/rules/usage.yaml:24]: https://github.com/gitpod-io/runbooks/pull/388

File: operations/observability/mixins/meta/rules/usage.yaml [hunk 2]
```diff
@@ -14,13 +14,24 @@ spec:
...
+        summary: There are failed usage reconciliations.
+        description: We have accumulated {{ printf "%.2f" $value }} failures. This affects how up-to-date usage data is.
+
+    - alert: GitpodUsageReconcileInvoicesFailures'
+      expr: sum(increase(grpc_server_handled_total{grpc_service="usage.v1.BillingService", grpc_method="ReconcileInvoices", grpc_code!="OK"})) > 1
+      for: 30m
```
Comment (andrew-farries) [inline operations/observability/mixins/meta/rules/usage.yaml:30]: I'm relatively new to PromQL, but what does the `for: 30m` do here?

Comment (easyCZ) [reply operations/observability/mixins/meta/rules/usage.yaml:30]: It says that the `expr` above needs to "have values" - be firing for 30m to fire an alert

Comment (andrew-farries) [reply operations/observability/mixins/meta/rules/usage.yaml:30]: So does that mean we alert when reconciliation fails for 30 minutes continuously, ie just one failure won't alert? (not sure how often we run reconciliation in prod currently.)

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**